### DISCUSSION TITLE
New Feature Idea

### DISCUSSION THREAD

**Discussion by ClosetCreeper** [ScratchAddons/ScratchAddons#4518] [Ideas]
In scratch under costumes there is a next costume block, could you make a last costume block that goes back 1 costume?

**Comment by Endermaster3690** [comment]
what about the non-sa users?

**Reply by ClosetCreeper** [reply]
For non-sa users it could look like this:
<img width="386" alt="Screen Shot 2022-04-19 at 7 56 11 PM" src="https://user-images.githubusercontent.com/95090991/164131445-cac33798-661e-4469-88e5-d0bdd6f14fad.png">

**Reply by ClosetCreeper** [reply]
But for sa-users, it could be a new "last costume" block.

**Reply by CST1229** [reply]
Or `switch costume to (join [previous costume] [])` (yes it works).
Or a "hacked" switch costume dropdown.

**Reply by kccuber-scratch** [reply]
> Or `switch costume to (join [previous costume] [])` (yes it works). Or a "hacked" switch costume dropdown.

Yeah that would work, we've done it for the `<key [ v] pressed>` block.

**Reply by ClosetCreeper** [reply]
I am confused, you added the feature?

**Comment by Endermaster3690** [comment]
The workaround is too easy.

**Reply by ClosetCreeper** [reply]
I use it so much, do you think it is possible to still make?

**Reply by Endermaster3690** [reply]
I'm not making it but I'm talking about it

**Reply by ClosetCreeper** [reply]
What do you mean? Does that mean that this feature idea is declined?

**Reply by Endermaster3690** [reply]
No, it's up to the devs. I'm saying I'm not a dev.

**Comment by BarthdryZ** [comment]
It is unneccessary and also the workaround appears to be easy. But a cool idea for mods.

**Comment by shock59** [comment]
It seems too niche considering how easy it is to do it manually, but you could always make the addon yourself and make a pull request if you really need it.

**Comment by mxmou** [comment]
There could be an addon that adds common combinations of blocks to the palette, including "forever if" and the workaround for this one.
